lib/arrays.md

```diff
@@ -5,44 +5,52 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
+If you try to retrieve an element that does not exist, it comes back as nil, or nothing.
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+If you search for a negative element, such as -6, you will be given the 6th element from the end, not the beginning, as the negative inverses the search. The reason -6 is the 6th is because when using the inverse, it starts at -1, as there is no -0.
+Therefore, -1 will always retrieve the last element.
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+In order to perform an operation on every element inside an array, you can loop through using the each method, which refers to each element inside it.
 
 # How do you add elements to an Array?
 
-Replace me with your response...
+To add an element to an array, use the push method. It will automatically add the element to the end of the array instead of to a new one. It literally pushes an element into an array.
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
 
-Replace me with your response...
+In order to replace something in an array, search through the array for the element using a while loop.
+The index is set at zero so that it starts at the first element. From there, have the original array. While the loop runs within the array, it searches for "Fioana" so that it can replace it with "Florence". Make sure it searches by an increment of one.
 
 # What do the methods `push`, `pop`, `shift`, and `unshift` do?
 
-Replace me with your response...
+Push: adds (or pushes) an element at the end of an array
+Pop: removes (or pops out) the last element of an array
+Shift: removes the first element of an array, causing the other elements to shift down in numerical position
+Unshift: adds objects to the front of an array, caushing the other elements to shift up in numerical position
 
 # How do you determine how many elements are in an Array?
 
-Replace me with your response...
+To find how many elements are in an array, use the length method (.length).
 
 # How would you reverse the order of elements in an Array?
 
-Replace me with your response...
+To reverse the order of elements in an array, use the reverse method (.reverse).
 
 # How would you convert a String `"Sumeet Jain"` into an Array `["Sumeet", "Jain"]`? How about to `["S", "u", "m", "e", "e", "t", " ", "J", "a", "i", "n"]`? How would you convert the Array back into a String?
 
-Replace me with your response...
\ No newline at end of file
+String to array: use split method. Split the string at the space. "Sumeet Jain".split" " results in ["Sumeet", "Jain"]
```

Comment:
Nice! How would you split up `"Sumeet Jain"` into `["S", "u", "m", "e", "e", "t", " ", "J", "a", "i", "n"]`?